staging/src/k8s.io/apiserver/pkg/storageversion/manager.go

```diff
@@ -267,10 +271,23 @@ func (s *defaultManager) Completed() bool {
 	return s.completed.Load().(bool)
 }
 
-func decodableVersions(e runtime.EquivalentResourceRegistry, gr schema.GroupResource) []string {
+func decodableVersions(directlyDecodableVersions []schema.GroupVersion, e runtime.EquivalentResourceRegistry, gr schema.GroupResource) []string {
 	var versions []string
+	for _, decodableVersions := range directlyDecodableVersions {
+		versions = append(versions, decodableVersions.String())
+	}
+
 	decodingGVRs := e.EquivalentResourcesFor(gr.WithVersion(""), "")
 	for _, v := range decodingGVRs {
+		found := false
+		for _, existingVersion := range versions {
+			if existingVersion == v.GroupVersion().String() {
+				found = true
+			}
+		}
+		if found {
```

Comment:
Do you have an example where `directlyDecodableVersions` contains versions that `decodingGVRs` doesn't have?

Reply:
> Do you have an example where `directlyDecodableVersions` contains versions that `decodingGVRs` doesn't have?

in https://github.com/kubernetes/kubernetes/pull/99840, simulating 1.22, apiextensions.k8s.io/v1beta1 is not listed as decodable by apiextensions/v1 without this change.